woocommerce/woocommerce-ios
# Fix 'imageEdgeInsets' was deprecated in iOS 15.0

Based on https://github.com/woocommerce/woocommerce-ios/pull/8094 , that PR should be merged first.

### Description
By updating the minimum release version to iOS 15 recently, we've seen some deprecation warnings. This PR fixes the case for `'imageEdgeInsets' was deprecated in iOS 15.0: This property is ignored when using UIButtonConfiguration`

On some buttons, we rely on the `distributeTitleAndImage(_:)` extension in order to add the correct padding between image and text in a UIButton. However, since we're moving to use `UIButton.Configuration` now, this functionality comes for free by calling `.imagePadding() ` instead. 

### Changes
- Deleted `UIButton+TitleAndImage`
- Added `.imagePadding()` usage to the 4 cases where we use this extension.

With the change, we can also change how we do other things, like setting title or image, but this is outside of the scope of this PR, which only intends to remove the Xcode deprecation warnings. We'll leave that for a different PR :) 

### Testing instructions
Confirm that the following buttons still keep the same separation between Image and Text:
- **Case 1:** Go to Orders > open an Order > Check the "+ Add Customer Note" and "+ "Add Note" buttons.
- **Case 2:** Go to Orders > open a "pending payment" Order > Tap on "Collect Payment" > Select "Card" > See the "Learn More about In-Person Payments" 
- **Case 3:** Go to Products > Tap on a Product > See the "+ Add more details" at the bottom of the view.

| Case 1 | Case 2 | Case 3 |
|-|-|-|
| <img width=400 src="https://user-images.githubusercontent.com/3812076/202135273-f1be770f-1cf9-46f3-867f-6f774e57ffd0.png"> | <img width=400 src="https://user-images.githubusercontent.com/3812076/202136475-ad5ca536-e338-4334-a543-880ccc6cc77c.png"> | <img width=400 src="https://user-images.githubusercontent.com/3812076/202136494-787b3e59-51fe-4f4a-8841-f09dfa217da6.png"> |
